src/rewrite.rs

```diff
@@ -37,6 +44,16 @@ impl<'a> RewriteContext<'a> {
             codemap: self.codemap,
             config: self.config,
             block_indent: self.block_indent + self.config.tab_spaces,
+            overflow_indent: self.overflow_indent,
+        }
+    }
+
+    pub fn overflow_context(&self, overflow: usize) -> RewriteContext<'a> {
```

Comment:
The intention with block indent is that it tracks only that component of indentation, as opposed to the offset/indent which we pass around which is the current indent. It is useful to carry round the block indent in case we want to block indent some part of a sub-expression.

E.g.,

```
let foo = match {
    bar => {
        foo();
    }
    baz => qux,
};

let foo2 =
    bar2();
```

In the first case we ignore the overflow, in the second case we use it. If we saved the overflow into the block indent, then we could not ignore it and match expressions would be over-indented. (Note that this case is easier because the overflow is the block indent, but it might be a visual indent, in which case there is no way to recover the block indent).